multiqc/modules/seqyclean/seqyclean.py

```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/env python3
+
+from __future__ import print_function
+from collections import OrderedDict
+import logging
+import re
+
+from multiqc import config
+from multiqc.plots import linegraph, bargraph, table
+from multiqc.modules.base_module import BaseMultiqcModule
+
+# Initialise the logger
+log = logging.getLogger(__name__)
+import traceback
+
+class MultiqcModule(BaseMultiqcModule):
+	def __init__(self):
+		# Initialise the parent object
+		super(MultiqcModule, self).__init__(name='Seqyclean', anchor='seqyclean',
+		href="http://www.awesome_bioinfo.com/my_module",
+		info="These graphs analyze the seqyclean files created during the pipeline. There are two associated graphs. The first one shows the total number of pairs kept and discarded. The second breaks down the reasons why certain pairs were discarded.")
+
+		# Parse logs
+		self.seqyclean = dict()
+		self.data_total = {}
+		self.data_breakdown = {}
+		for f in self.find_log_files('seqyclean'):
+			RowTwo = f['f'].split("\n")[1] #split the file into two rows
+			#These variables will be used in the total data table 
+			PairsKept = RowTwo.split("\t")[57] # get the number of pairs kept (from the second row)
+			PairsDiscarded = RowTwo.split("\t")[61] #get num of pairs discarded
+			#These are for the breakdown
+			PE1ReadsAn = RowTwo.split("\t")[21]
+			PE1TruSeqAdap_found = RowTwo.split("\t")[23]
+			PE1ReadswVector_found = RowTwo.split("\t")[25]
+			PE1ReadswContam_found = RowTwo.split("\t")[27]
+			PE1DiscByContam = RowTwo.split("\t")[37]
+			PE1DiscByLength = RowTwo.split("\t")[38]
+			PE2ReadsAn = RowTwo.split("\t")[39]
+			PE2TruSeqAdap_found = RowTwo.split("\t")[41]
+			PE2ReadswVector_found = RowTwo.split("\t")[43]
+			PE2ReadswContam_found = RowTwo.split("\t")[45]
+			PE2DiscByContam = RowTwo.split("\t")[55]
+			PE2DiscByLength = RowTwo.split("\t")[56]
```

Comment:
While this code probably works for now, it has a few of problems. The main one is that if the tool authors change anything (new columns, different column order etc), then everything breaks. Worse, it may continue working but giving the wrong data silently.

We can achieve the same thing by using the column names in the first row of the file. The code will be more robust and will fail loudly if anything more major changes.

Something like this should work I think: _(untested)_

```python
rows =  f['f'].splitlines()
headers = rows[0].split("\t")
cols = rows[1].split("\t")
for header, col in zip(headers, cols):
    data_total[f['s_name']][header] = col
```